main.go

```diff
@@ -428,7 +428,7 @@ func newCreateOrderCommand(db *sql.DB) *cli.Command {
 	}
 }
 
-func newShowCustomerCommand(db *sql.DB, ctx context.Context) *cli.Command {
+func newShowCustomerCommand(db **sql.DB, ctx context.Context) *cli.Command {
```

Comment:
The bug here is due to some tricky pointer and scoping issues - this was a tricky one. I'll try to explain what's going on here:

We first create the `*sql.DB` variable in main (let's call it variable **A**):
```
var db *sql.DB
```
We pass variable **A** when we call `newShowCustomerCommand(db, ctx)` in `main`. 

Next, the program executes `app.Before` where we create a new `*sql.DB` variable (let's call it variable **B**), and then we assign it to `db`. 
```go
Before: func(cCtx *cli.Context) error {
	...
	db, err = connectDB(org)
```

This is where things gets tricky. In the scope of main, the `db` variable is now pointing at variable **B**. 
However, the `db` variable within the scope of the `showCustomerCommand.Action` function is still variable **A**. 
We basically "need to go lower" in regards to pointers to make that variable reassignment we do in the `Before` function (`db, err = connectDB(org)`) also changes which variable we reference in the `showCustomerCommand.Action` scope. That's why we pass a `db **sql.DB` instead of a `db *sql.DB` now.

Reply:
For what it's worth, in a "real world" app we would most likely approach this problem by changing when/how we connect to the database. We could design this in a way where we don't need to think about this pointer logic nearly as much.